identity/src/main/java/com/android/identity/credential/NameSpacedData.kt

```diff
@@ -0,0 +1,386 @@
+        }
+
+        /**
+         * Encode the given value as an integer or unsigned integer and adds it to the builder.
+         *
+         * @param nameSpaceName the name space name.
+         * @param dataElementName the data element name.
+         * @param value The value, as a `long`.
+         * @return the builder.
+         */
+        fun putEntryNumber(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: Long
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeNumber(value))
+        }
+
+        /**
+         * Encode the given value as a boolean and adds it to the builder.
+         *
+         * @param nameSpaceName the name space name.
+         * @param dataElementName the data element name.
+         * @param value The value, as a `boolean`..
+         * @return the builder.
+         */
+        fun putEntryBoolean(
+            nameSpaceName: String,
+            dataElementName: String,
+            value: Boolean
+        ): Builder {
+            return putEntry(nameSpaceName, dataElementName, Util.cborEncodeBoolean(value))
+        }
+
+        /**
+         * Builds a [NameSpacedData] from the builder
+         *
+         * @return a [NameSpacedData] instance.
+         */
+        fun build(): NameSpacedData {
+            return NameSpacedData(map)
+        }
+    }
+
+    companion object {
+        /**
+         * Creates a new [NameSpacedData] from encoded CBOR.
+         *
+         * @param encodedCbor CBOR encoded in the format described by [.encodeAsCbor].
+         * @return A [NameSpacedData].
+         * @throws IllegalArgumentException if the given data is not valid CBOR.
+         * @throws IllegalArgumentException if the given data does not confirm to the CDDL in
+         * [.encodeAsCbor].
+         */
+        @JvmStatic
+        fun fromEncodedCbor(encodedCbor: ByteArray): NameSpacedData {
+            return fromCbor(Util.cborDecode(encodedCbor))
+        }
+
+        private fun fromCbor(dataItem: DataItem): NameSpacedData {
+            val ret = LinkedHashMap<String, LinkedHashMap<String, ByteArray>>()
+            require(dataItem is Map) { "dataItem is not a map" }
+            for (nameSpaceNameItem in dataItem.keys) {
+                require(nameSpaceNameItem is UnicodeString) { "Expected string for namespace name" }
+                val namespaceName: String = nameSpaceNameItem.string
+                val dataElementToValueMap = LinkedHashMap<String, ByteArray>()
+                val dataElementItems: DataItem = Util.cborMapExtractMap(dataItem, namespaceName)
+                require(dataElementItems is Map) { "Expected map" }
+                for (dataElementNameItem in dataElementItems.keys) {
+                    require(dataElementNameItem is UnicodeString) { "Expected string for data element name" }
+                    val dataElementName: String = dataElementNameItem.string
```

Comment:
no need to explicitly define type `String`

`val dataElementName: String = dataElementNameItem.string`